### PR TITLE
[PR-TEMPLATE] Template for PR consistency

### PR DESCRIPTION
**Pull request by gfunk1230**
Create a PR template we can use on every PR we create. Please feel free to comment and contribute here.

Please don't feel we need to keep all the sections or checkboxes. I added a lot purposely in order to give us room to pick and choose what makes the most sense to us. 

image

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+### PR Title Example: [DEVHUB-1316] Create a script to refresh staging data with prod data
+
+## Jira Ticket:
+
+Link to jira ticket [DEVHUB-1111]
+
+Example: [DEVHUB-1316](https://jira.mongodb.org/browse/DEVHUB-1316)
```
Comment (ajhenri) [inline PULL_REQUEST_TEMPLATE.md:7]: Do we need this example here? I think it is straightforward to copy and paste the link. I've found when working with the old template that I would spend a second deleting this before putting in the ticket.

Comment (gfunk1230) [reply PULL_REQUEST_TEMPLATE.md:7]: @ajhenri Yeah I agree was just being explicit but I will remove it.

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+## Checklist: (Check off where applicable)?
+
+-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
+-   [ ] Have you reviewed with Design or Product (If Applicable)?
+-   [ ] I have performed a self-review(QA) of my code
+-   [ ] I have commented my code, particularly in hard-to-understand areas
+-   [ ] I have made corresponding changes to the documentation
+-   [ ] I have added tests that prove my fix is effective or that my feature works
+-   [ ] New and existing unit tests pass locally with my changes
+-   [ ] Any dependent changes have been merged and published in downstream modules
```
[the remark this answers is not in the document]
Comment (mcolella14) [reply PULL_REQUEST_TEMPLATE.md:24]: Ahh ok cool cool, I like these 

-  I have made corresponding changes to the documentation
- I have commented my code, particularly in hard-to-understand areas
- I have added tests that prove my fix is effective or that my feature works

Most of the others seem kinda redundant or irrelevant, and the Design/Product one is good but that step would probably be after opening a PR.

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+### PR Title Example: [DEVHUB-1316] Create a script to refresh staging data with prod data
```
Comment (ajhenri) [inline PULL_REQUEST_TEMPLATE.md:1]: I would remove this as well. There is a note in the README.md on PR titles https://github.com/mongodb/devcenter#development-and-deployment-workflow. We can enforce that so we don't have to put it in the template.

**Review by ajhenri — APPROVED** [review]
Aside from above comments, LGTM!

File: PULL_REQUEST_TEMPLATE.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gfunk1230) [reply PULL_REQUEST_TEMPLATE.md:24]: Thanks @mcolella14

**Review by mcolella14 — APPROVED** [review]

**Pull request merged**